detection_rules/rule.py

```diff
@@ -215,20 +215,29 @@ class FlatThreatMapping(MarshmallowDataclassMixin):
     sub_technique_ids: List[str]
 
 
+@dataclass(frozen=True)
+class AlertSuppressionDuration:
+    """Mapping to alert suppression duration."""
+    unit: definitions.TimeUnits
+    value: definitions.AlertSuppressionValue
+
+
 @dataclass(frozen=True)
 class AlertSuppressionMapping(MarshmallowDataclassMixin, StackCompatMixin):
     """Mapping to alert suppression."""
-    @dataclass
-    class AlertSuppressionDuration:
-        """Mapping to allert suppression duration."""
-        unit: definitions.TimeUnits
-        value: int
 
-    group_by: List[definitions.NonEmptyStr]
+    group_by: definitions.AlertSuppressionGroupBy
     duration: Optional[AlertSuppressionDuration]
     missing_fields_strategy: definitions.AlertSuppressionMissing
 
 
+@dataclass(frozen=True)
+class ThresholdAlertSuppression:
+    """Mapping to alert suppression."""
+
+    duration: Optional[AlertSuppressionDuration]
```

Comment:
not sure if the property of this class should match schema 1 to 1,
but according to https://www.elastic.co/guide/en/security/current/rules-api-create.html#_threshold_rule, 
`duration` is required for alert suppression in threshold rule

Reply:
Thanks @vitaliidm great catch.